plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergSmoke.java

```diff
@@ -1027,6 +1028,55 @@ public void testTruncateTextTransform()
         dropTable("test_truncate_text_transform");
     }
 
+    @Test(dataProvider = "truncateNumberTypesProvider")
+    // This particular method may or may not be @Flaky. It is annotated since the problem is generic.
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/5201", match = "Failed to read footer of file: HdfsInputFile")
+    public void testTruncateNumberTransform(String dataType)
+    {
+        String table = format("test_truncate_%s_transform", dataType.toLowerCase(ENGLISH));
+        String select = "SELECT d_trunc, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"" + table + "$partitions\"";
+
+        assertUpdate(format("CREATE TABLE " + table + " (d %s, b BIGINT) WITH (partitioning = ARRAY['truncate(d, 10)'])", dataType));
+
+        @Language("SQL") String insertSql = "INSERT INTO " + table + " VALUES" +
+                "(0, 1)," +
+                "(1, 2)," +
+                "(5, 3)," +
+                "(9, 4)," +
+                "(10, 5)," +
+                "(11, 6)," +
+                "(-1, 7)," +
+                "(-5, 8)," +
+                "(-10, 9)," +
+                "(-11, 10)";
+        assertUpdate(insertSql, 10);
+
+        assertQuery("SELECT COUNT(*) FROM \"" + table + "$partitions\"", "SELECT 4");
+
+        assertQuery("SELECT b FROM " + table + " WHERE d - (((d % 10) + 10) % 10) = 0", "SELECT b FROM (VALUES (1), (2), (3), (4)) AS t(b)");
+        assertQuery(select + " WHERE d_trunc = 0", "VALUES(0, 4, 0, 9, 1, 4)");
+
+        assertQuery("SELECT b FROM " + table + " WHERE d - (((d % 10) + 10) % 10) = 10", "SELECT b FROM (VALUES (5), (6)) AS t(b)");
+        assertQuery(select + " WHERE d_trunc = 10", "VALUES(10, 2, 10, 11, 5, 6)");
+
+        assertQuery("SELECT b FROM " + table + " WHERE d - (((d % 10) + 10) % 10) = -10", "SELECT b FROM (VALUES (7), (8), (9)) AS t(b)");
+        assertQuery(select + " WHERE d_trunc = -10", "VALUES(-10, 3, -10, -1, 7, 9)");
+
+        assertQuery("SELECT b FROM " + table + " WHERE d - (((d % 10) + 10) % 10) = -20", "SELECT b FROM (VALUES (10)) AS t(b)");
+        assertQuery(select + " WHERE d_trunc = -20", "VALUES(-20, 1, -11, -11, 10, 10)");
+
+        dropTable(table);
+    }
+
+    @DataProvider
+    public Object[][] truncateNumberTypesProvider()
+    {
+        return new Object[][] {
+                {"INT"},
+                {"BIGINT"},
```

Comment:
- lowercase them (we strive to write types in lowercase (except literals like `BIGINT '42'`), and SQL keywords in uppercase)
- add tinyint, smallint ?